src/borda_clj/client.clj

```diff
@@ -0,0 +1,60 @@
+(ns borda-clj.client
+  (:use [slingshot.slingshot :only [throw+ try+]])
+  (:require [aleph.http :as http]
+            [byte-streams :as bs]
+            [cheshire.core :as json]))
+
+(defn collect
+  "Adds the given measurement (values and dimensions) to the given hashmap and caps the size of the hashmap at the given max-buffer-size"
+  [measurements max-buffer-size dimensions values]
+  (let [key (merge (sorted-map) dimensions)]
+    (if (contains? measurements key)
+      ; TODO: support stuff other than SUM here (e.g. AVG, MIN, MAX, etc.)
+      (assoc measurements key (merge-with + (get measurements key) values)) ; merge with existing in buffer
+      (if (< (count measurements) max-buffer-size)
+        (assoc measurements key values) ; space available, add to buffer
+        (do (print "borda buffer full, discarding measurement" dimensions values) measurements))))) ; buffer full
```

Comment:
I would put `measurements` on its own line to highlight that it's the returned value.

Reply:
Will do.